ecosystem/indexer-grpc/indexer-grpc-utils/src/lib.rs

```diff
@@ -24,3 +39,39 @@ pub fn get_file_store_bucket_name() -> String {
 pub fn get_health_check_port() -> u16 {
     std::env::var("HEALTH_CHECK_PORT").map_or_else(|_| 8080, |v| v.parse::<u16>().unwrap())
 }
+
+pub async fn update_cache_latest_version(
+    conn: &mut impl redis::ConnectionLike,
+    latest_version: u64,
+) -> Result<(), Box<dyn std::error::Error>> {
+    conn.set(CACHE_LATEST_VERSION, latest_version)
+        .map_err(|e| e.into())
+}
+
+pub async fn get_cache_coverage_status(
+    conn: &mut impl redis::ConnectionLike,
+    requested_version: u64,
+) -> Result<CacheCoverageStatus, Box<dyn std::error::Error>> {
+    let latest_version: u64 = match conn.get(CACHE_LATEST_VERSION) {
+        Ok(v) => v,
+        Err(err) => return Err(err.into()),
+    };
+    if requested_version + BLOB_STORAGE_SIZE > latest_version {
+        Ok(CacheCoverageStatus::DataNotReady)
+    } else if requested_version + CACHE_SIZE_ESTIMATION < latest_version {
+        Ok(CacheCoverageStatus::CacheEvicted)
+    } else {
+        Ok(CacheCoverageStatus::CacheHit)
+    }
```

Comment:
id' rewrite this code in a way that's clearer (as we discussed in person)

Reply:
done